app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmReceiver.java

```diff
@@ -47,22 +44,20 @@ public void onReceive(Context context, Intent intent) {
             String title = intent.getStringExtra(BundleKeys.ALARM_TITLE);
             //Toast.makeText(context, "Alarm worked.", Toast.LENGTH_LONG).show();
 
-            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
-            boolean defaultValue = context.getResources().getBoolean(R.bool.preferences_insistent_alarm_enabled_default_value);
-            boolean insistent = prefs.getBoolean("insistent", defaultValue);
+            AppRepository appRepository = AppRepository.INSTANCE;
 
             Intent launchIntent = MainActivity.createLaunchIntent(context, sessionId, day);
             PendingIntent contentIntent = PendingIntent
                     .getActivity(context, lid, launchIntent, PendingIntent.FLAG_ONE_SHOT);
 
             NotificationHelper notificationHelper = new NotificationHelper(context);
-            String defaultReminderTone = context.getString(R.string.preferences_reminder_tone_default_value);
-            Uri soundUri = Uri.parse(prefs.getString("reminder_tone", defaultReminderTone));
+            String alarmToneUriString = appRepository.readAlarmTone();
+            Uri soundUri = Uri.parse(alarmToneUriString);
```

Comment:
It looks like you do this at all call sites of `AppRepository.readAlarmTone()`. Why not return a `Uri` from the repository?

Reply:
I agree. Changed in 0ea7501fe86c818c94ee3f33b47e3d86175865da.

Reply:
I agree. Changed in 0ea7501fe86c818c94ee3f33b47e3d86175865da.